projects/example-app-cypress/integration/round-trip.spec.ts

```diff
@@ -1,14 +1,34 @@
+let LOCAL_STORAGE_MEMORY = {};
+
+Cypress.Commands.add('saveLocalStorage', () => {
+  Object.keys(localStorage).forEach(key => {
+    LOCAL_STORAGE_MEMORY[key] = localStorage[key];
+  });
+});
+
+Cypress.Commands.add('restoreLocalStorage', () => {
+  Object.keys(LOCAL_STORAGE_MEMORY).forEach(key => {
+    localStorage.setItem(key, LOCAL_STORAGE_MEMORY[key]);
+  });
+});
+
 context('Full round trip', () => {
   before(() => {
     (cy as any).eyesOpen({
       appName: 'books_app',
       testName: 'round-trip',
       browser: { width: 800, height: 600 },
     });
-    window.indexedDB.deleteDatabase('books_app');
+    window.localStorage.removeItem('books_app');
     cy.visit('/');
   });
+  beforeEach(() => {
```

Comment:
Is the `beforeEach` and `afterEach` hook needed?

Reply:
Required to persist the `localStorage` between the tests, if not we would have to duplicate test code to add books to the collection as `cypress` clears the `localStorage` after each test case